src/libPMacc/include/cuSTL/cursor/tools/twistVectorFieldAxes.hpp

```diff
@@ -55,18 +61,59 @@ struct TwistVectorFieldAxes
  *
  * e.g.: new_cur = twistVectorFieldAxes<math::CT::Int<1,2,0> >(cur); // x -> y, y -> z, z -> x
  *
- * \tparam Axes compile-time vector (PMacc::math::CT::Int) that describes the mapping.
- * x-axis -> Axes::at<0>, y-axis -> Axes::at<1>, ...
+ * \tparam T_Permutation compile-time vector (PMacc::math::CT::Int) that describes the mapping.
+ * x-axis -> T_Permutation::at<0>, y-axis -> T_Permutation::at<1>, ...
+ *
+ */
+template<typename T_Permutation, typename T_Cursor>
+HDINLINE
```

Comment:
The order is changed that it is not allowed to call this method like `twistVectorFieldAxes<CT::Int<1,1> >(cursor, permutationWithWrongType)`
The order for the case that you need to give the permutation as compile time signature is still the same as the old.